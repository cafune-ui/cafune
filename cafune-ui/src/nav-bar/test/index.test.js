import NavBar from '../';
import { render, mount } from 'enzyme';
import { shallow, deep } from 'preact-render-spy';

describe('NavBar', () => {
  it('should render properly', () => {
    const backFn = jest.fn();
    const searchFn = jest.fn();
    const wrapper = mount(
      <NavBar
        isFixed
        left={{
          icon: 'wrong',
          text: '返回',
          color: '#369',
          clickHandler: backFn
        }}
        rights={{
          icon: 'search',
          text: '搜索',
          badge: '22',
          clickHandler: searchFn
        }}
      >
        标题
      </NavBar>
    );
    expect(wrapper).toMatchSnapshot();
      
    console.log( wrapper.find('.caf-navbar__btns--left .caf-navbar__btn'))
    wrapper.find('.caf-navbar__btns--left .caf-navbar__btn').simulate('click');
    expect(backFn).toHaveBeenCalled();
    wrapper.find('.caf-navbar__btns--right .caf-navbar__btn').simulate('click');
    expect(searchFn).toHaveBeenCalled();

    const wrapper2 = deep(
      <NavBar
        isFixed
        left={{
          text: '返回',
          color: '#369',
          badge: true,
          clickHandler: backFn
        }}
        rights={[
          {
            icon: 'search',
            text: '搜索',
            clickHandler: searchFn
          },
          null
        ]}
      >
        标题
      </NavBar>
    );
    expect(wrapper2).toMatchSnapshot();
  });
});
