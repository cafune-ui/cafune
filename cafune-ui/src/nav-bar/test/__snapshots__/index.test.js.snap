// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render properly 1`] = `
RenderContext {
  "0": VNode {
    "attributes": undefined,
    "children": Array [],
    "key": undefined,
    "nodeName": [Function],
  },
  "length": 1,
}
`;
